lrauv_ignition_plugins/src/ScienceSensorsSystem.cc

```diff
@@ -99,13 +118,36 @@ class tethys::ScienceSensorsSystemPrivate
   public: std::string dataPath {"2003080103_mb_l3_las.csv"};
 
   /// \brief Indicates whether data has been loaded
-  public: bool initialized = false;
+  public: bool initialized {false};
+
+  /// \brief Set to true after the spherical coordinates have been initialized.
+  /// This may happen at startup if the SDF file has them hardcoded, or at
+  /// runtime when the first vehicle is spawned. Assume the coordinates are
+  /// only shifted once.
+  public: bool worldSphericalCoordsInitialized {false};
+
+  /// \brief Mutex for writing to world origin association to lat/long
+  public: std::mutex mtx;
+
+  /// \brief Spherical coordinates of world origin. Can change at any time.
+  public: ignition::math::SphericalCoordinates worldOriginSphericalCoords;
+
+  /// \brief World origin in spherical position (latitude, longitude,
+  /// elevation), angles in degrees
+  public: ignition::math::Vector3d worldOriginSphericalPos = {0, 0, 0};
+
+  /// \brief World origin in Cartesian coordinates converted from spherical
+  /// coordinates
+  public: ignition::math::Vector3d worldOriginCartesianCoords = {0, 0, 0};
```

Comment:
I don't understand this variable. Isn't the world origin in Cartesian always zero? What does a non-zero cartesian origin mean?